st2common/st2common/transport/bootstrap_utils.py

```diff
@@ -69,3 +72,12 @@ def wrapped_register_exchanges(connection, channel):
                                       retry_wrapper=retry_wrapper)
 
         retry_wrapper.run(connection=conn, wrapped_callback=wrapped_register_exchanges)
+
+
+def register_exchanges_with_retry():
+    retrying_obj = retrying.Retrying(
+        retry_on_exception=socket.error,
+        wait_fixed=cfg.CONF.messaging.connection_retry_wait,
+        stop_max_attempt_number=cfg.CONF.messaging.connection_retries
```

Comment:
The main issue boils down to [this line](https://github.com/StackStorm/st2/blob/6a530bc5a6053d8cfd36b61903af3f2845c7bdbd/st2common/st2common/transport/bootstrap_utils.py#L65). The retry logic is backed in below this line; however, the connection is made on line 65. If rabbitmq server is down when that line is executed it will fail without retry.

As for the config options I tend to agree that those options should be brought together. I was unaware that they exist. I would be more in favor of moving `connection_retry_wrapper`'s properties to the config file since those properties seem like things that should be encouraged to modify.